src/ddspdrum/module.py

```diff
@@ -58,6 +62,94 @@ def control_to_sample_rate(self, control: np.array) -> np.array:
             )
             return resample(control, num_samples)
 
+    def add_parameter(self, parameter_id: str, value: float, minimum: float,
+                      maximum: float, scale: float = 1):
+        """
+        Add a new parameter to this module.
+
+        Parameters
+        ----------
+        parameter_id (str)  :   id to save this parameter as
+        value (float)       :   initial value for this parameter
+        minimum (float)     :   minimum value that this parameter can take
+        maximum (float)     :   maximum value that this parameter can take
+        scale (float)       :   scaling when converting between [0,1] range
+        """
+        if parameter_id in self.parameters:
+            raise ValueError("parameter_id: {} already used".format(parameter_id))
+
+        self.parameters[parameter_id] = Parameter(value, minimum, maximum,
```

Comment:
Perhaps we could simplify and in each `SynthModule.__init__` it calls ` Parameter(` multiple times instead of `add_parameter`? Then we also don't have to duplicate the documentation and maintain it twice.

The code to sanity-check we don't duplicate parameters (and that params are in the right range, etc.), I think we should maybe write generically as `SynthModule.sanity_check` and `SynthModule.__call__` will always run sanity-check before call the instance `ActualModule.__call_implementation__` or something like that.

Reply:
Sure, so just add Parameters directly to self.parameters in `__init__`? 

Not sure if I totally understand how the sanity check function would work.

Reply:
@jorshi don't worry about the sanity check, I could implement it in another PR.

Yeah the simple idea is that, by convention, all __init__ directly create `Parameters`. This traces a nice clean line to a PyTorch port where every `nn.Module` will init its torch parameters in `__init__`. If you're not sure about this I can tackle it.

Reply:
Implemented here: https://github.com/turian/ddsp-drum/pull/21